operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/Controller.java

```diff
@@ -195,7 +195,7 @@ public DeleteControl execute() throws Exception {
 
               // The cleanup is called also when explicit invocation is true, but the cleaner is not
               // implemented
-              if (managedWorkflow.hasCleaner() || !explicitWorkflowInvocation) {
+              if (managedWorkflow.hasCleaner() && !explicitWorkflowInvocation) {
```

Comment:
TBH, I miss the context here, but if I understand the comment above correctly it should be:
```suggestion
              if (managedWorkflow.hasCleaner() || explicitWorkflowInvocation) {
```

Reply:
Updated the PR and also the comment. It's a bit more complicated than that.